INPTP_AppForFixing/EmployeeDialog.cs

```diff
@@ -1,59 +1,78 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.ComponentModel;
-using System.Data;
-using System.Drawing;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using System.Windows.Forms;
 
 namespace INPTP_AppForFixing
 {
     public partial class EmployeeDialog : Form
     {
         private MainForm main;
-        private bool employee=false;
+        private bool employee = false;
 
-        public EmployeeDialog(MainForm main, bool employee)
+        public EmployeeDialog(MainForm main, bool employee, Boss boss = null)
         {
             InitializeComponent();
-            this.main = main;
-            this.employee = employee;
-            if (employee) {
-                lbDep.Visible = false;
-                lbDpName.Visible = false;
-                tBDpName.Visible = false;
+            Init(main, employee);
+
+            if (boss != null)
+            {
+                tBID.Text = boss.Id.ToString();
+                tBID.Enabled = false;
+                tBFirstName.Text = boss.FirstName;
+                tBLastName.Text = boss.LastName;
+                tBJob.Text = boss.Job;
+                tBSalary.Text = boss.MonthlySalaryCZK.ToString();
+                dateTimePickerBirthDate.Value = boss.OurBirthDate;
             }
         }
 
         private void btnOk_Click(object sender, EventArgs e)
         {
             if (testOfValues())
             {
-                main.Bosses.Add(new Boss(Int32.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text), new Department(tBDpName.Text)));
+                int Id = Int32.Parse(tBID.Text);
+                Boss newB = new Boss(Id, tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text), new Department(tBDpName.Text));
+                Boss b = main.Bosses.FirstOrDefault(x => x.Id == Id);
+                if (b == null)
+                {
+                    main.Bosses.Add(newB);
+                }
+                else
+                {
+                    main.Bosses.RemoveWhere(x => x.Id == newB.Id);
+                    main.Bosses.Add(newB);
+                    b = newB;
+                }
                 main.OnBossesChange();
-                Close();
+                this.Close();
             }
-            else {
-                
-                    MessageBox.Show("Bad entry data inserted.", "Error",
-                                    MessageBoxButtons.OK, MessageBoxIcon.Error);
+            else
+            {
+                MessageBox.Show("Bad entry data inserted.", "Error",
+                                MessageBoxButtons.OK, MessageBoxIcon.Error);
             }
         }
 
-        private void EmployeeDialog_FormClosed(object sender, FormClosedEventArgs e)
+        private bool testOfValues()
```

Comment:
test of what values and what are you testing? name of this method could be much better...

Reply:
Fixed - renamed to correct name (ValidFormInput)